src/base/hasher.hpp

```diff
@@ -21,12 +21,13 @@
 #define BUILDCACHE_HASHER_HPP_
 
 #include <cstdint>
+
 #include <map>
+#include <stdexcept>
 #include <string>
 
-extern "C" {
-#include <md4/md4.h>
-}
+#define XXH_INLINE_ALL
```

Comment:
I'm a bit worried by this. I suspect that for our use cases we will not see any speedup by inlining the code (we're usually dealing with large blocks of unknown size). OTOH I saw a binary size increase of 400KB (buildcache executable under Linux), compared to the master branch with MD4, which can hurt performance in other ways (e.g. disk & CPU caches).

I wonder if the buildcache executable would be smaller without `XXH_INLINE_ALL`.

Reply:
**Update 2**: After trying harder, I was able to reduce the binary size with some 360KB. See this draft: https://github.com/mbitsnbites/buildcache/pull/141 I have not benchmarked it to see if it performs worse or not, though.